cli/cmd/compose/up.go

```diff
@@ -201,6 +215,38 @@ func runCreateStart(ctx context.Context, opts upOptions, services []string) erro
 	return err
 }
 
+func applyScale(opts []string, project *types.Project) error {
+SCALE:
+	for _, scale := range opts {
+		split := strings.Split(scale, "=")
+		if len(split) != 2 {
+			return fmt.Errorf("invalid --scale option %q. Should be SERVICE=NUM", scale)
+		}
+		name := split[0]
+		replicas, err := strconv.Atoi(split[1])
+		if err != nil {
+			return err
+		}
+		for i, s := range project.Services {
+			if s.Name == name {
+				service, err := project.GetService(name)
+				if err != nil {
+					return err
+				}
+				if service.Deploy == nil {
+					service.Deploy = &types.DeployConfig{}
+				}
+				count := uint64(replicas)
+				service.Deploy.Replicas = &count
+				project.Services[i] = service
+				continue SCALE
```

Comment:
errf... I will need to spend some time to look for a nicer solution, but ok (with some tests around it for now)

Reply:
I like using `for` labels for such scenarios. The other ways are either define sub-functions that would hide this `goto` by doing `return` inside the loop, or include a `var found bool` as a guard.